nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/client/ESRestClient.java

```diff
@@ -0,0 +1,318 @@
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.search.ClearScrollRequest;
+import org.elasticsearch.action.search.ClearScrollResponse;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.search.SearchScrollRequest;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.cluster.health.ClusterHealthStatus;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.nuxeo.elasticsearch.api.ESClient;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * @since 9.3
+ */
+public class ESRestClient implements ESClient {
+    private static final org.apache.commons.logging.Log log = LogFactory.getLog(ESRestClient.class);
+    protected RestClient lowLevelClient;
+    protected RestHighLevelClient client;
+
+    public ESRestClient(RestClient lowLevelRestClient, RestHighLevelClient client) {
+        this.lowLevelClient = lowLevelRestClient;
+        this.client = client;
+    }
+
+    @Override
+    public boolean waitForYellowStatus(String[] indexNames, int timeoutSecond) {
+        ClusterHealthStatus healthStatus;
+        Response response;
+        try {
+            response = lowLevelClient.performRequest("GET",
+                    String.format("/_cluster/health/%s?wait_for_status=yellow&timeout=%ds",
+                            getIndexesAsString(indexNames), timeoutSecond));
+            try (InputStream is = response.getEntity().getContent()) {
+                Map<String, Object> map = XContentHelper.convertToMap(XContentType.JSON.xContent(), is, true);
+                healthStatus = ClusterHealthStatus.fromString((String) map.get("status"));
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        switch (healthStatus) {
+            case GREEN:
+                log.info("ES Cluster ready: " + response);
+                return true;
+            case YELLOW:
+                log.warn("Es Cluster ready but not GREEN: " + response);
+                return false;
+            default:
+                String error = "ES Cluster health status: " + healthStatus + ", not Yellow after " + timeoutSecond + " give up: "
+                        + response;
+                log.error(error);
+                throw new IllegalStateException(error);
+        }
+    }
+
+    protected String getIndexesAsString(String[] indexNames) {
+        String indexes = "";
+        if (indexNames != null && indexNames.length > 0) {
+            indexes = String.join(",", indexNames);
+        }
```

Comment:
Don't do double assignment, and no need for "return at end of method", and the case `indexNames.length == 0` is already dealt with by `String.join`, so just do:

    return indexNames == null ? "" : String.join(",", indexNames);